src/pack_encrypted.rs

```diff
@@ -0,0 +1,176 @@
+    ///    the recipient is not known in advance (e.g., in a broadcast scenario).
+    ///    Adding a signature when one is not needed can degrade rather than enhance security because
+    ///    it relinquishes the sender’s ability to speak off the record.
+    /// - `did_resolver` instance of `DIDResolver` to resolve DIDs.
+    /// - `secrets_resolver` instance of SecretsResolver` to resolve sender DID keys secrets.
+    /// - `options` allow fine configuration of packing process and have implemented `Default`.
+    ///
+    /// # Returns
+    /// Tuple `(encrypted_message, metadata)`.
+    /// - `encrypted_message` A DIDComm encrypted message as a JSON string.
+    /// - `metadata` additional metadata about this `pack` execution like used keys identifiers,
+    ///   used messaging service.
+    ///
+    /// # Errors
+    /// - `DIDNotFound` Sender or recipient DID or DID key is not found.
+    /// - `SecretNotFound` Sender secret is not found.
+    /// - `NoCompatibleCrypto` No compatible keys are found between sender and recipient.
+    /// - `InvalidState` Indicates library error.
+    /// - `IOError` IO error during DID or secrets resolving
+    /// TODO: verify and update errors list
+    pub async fn pack_encrypted<'dr, 'sr>(
+        &self,
+        _to: &str,
+        _from: Option<&str>,
+        _from_signed: Option<&str>,
+        _did_resolver: &'dr (dyn DIDResolver + 'dr),
+        _secrets_resolver: &'sr (dyn SecretsResolver + 'sr),
+        _options: &PackEncryptedOptions,
+    ) -> Result<(String, PackEncryptedMetadata)> {
+        todo!("Implement me.");
+    }
+}
+
+/// Allow fine configuration of packing process.
+pub struct PackEncryptedOptions {
+    /// If `true` and message is authenticated than information about sender will be hidden from mediators, but
+    /// additional re-encryption will be required. For anonymous messages this property will be ignored.
+    pub hide_sender: bool,
+
+    /// Whether the encrypted messages need to be wrapped into `Forward` messages to be sent to Mediators
+    /// as defined by the `Forward` protocol.
+    pub forward: bool,
+
+    /// if forward is enabled these optional headers can be passed to the wrapping `Forward` messages.
+    /// If forward is disabled this property will be ignored.
+    pub forward_headers: Option<Vec<(String, Value)>>,
+
+    /// Identifier (DID URL) of messaging service (https://identity.foundation/didcomm-messaging/spec/#did-document-service-endpoint).
+    /// If DID contains multiple messaging services it allows specify what service to use.
+    /// If not present first service will be used.
+    pub messaging_service: Option<String>,
+
+    /// Algorithm used for authenticated encryption
+    pub enc_alg_auth: AuthCryptAlg,
+
+    /// Algorithm used for anonymous encryption
+    pub enc_alg_anon: AnonCryptAlg,
+}
+
+impl Default for PackEncryptedOptions {
+    fn default() -> Self {
+        PackEncryptedOptions {
+            hide_sender: false,
```

Comment:
I think `protect_sender_id` is better as it matches the spec: https://identity.foundation/didcomm-messaging/spec/#protecting-the-sender-identity

Reply:
done. Thanks.